ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java

```diff
@@ -0,0 +1,138 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.model.InstanceConfig;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * A class with clustermap related utility methods for use by other classes.
+ */
+public class ClusterMapUtils {
+  static final String DISK_CAPACITY_STR = "capacityInBytes";
+  static final String DISK_STATE = "diskState";
+  static final String REPLICAS_STR = "Replicas";
+  static final String REPLICAS_DELIM_STR = ",";
+  static final String REPLICAS_STR_SEPARATOR = ":";
+  static final String SSLPORT_STR = "sslPort";
+  static final String RACKID_STR = "rackId";
+  static final String SEALED_STR = "SEALED";
+  static final String ONLINE_STR = "ONLINE";
+  static final String OFFLINE_STR = "OFFLINE";
+  static final String ZKCONNECTSTR_STR = "zkConnectStr";
+  static final String ZKINFO_STR = "zkInfo";
+  static final String DATACENTER_STR = "datacenter";
+  static final int UNKNOWN_RACK_ID = -1;
+  static final int MIN_PORT = 1025;
+  static final int MAX_PORT = 65535;
+  static final long MIN_REPLICA_CAPACITY_IN_BYTES = 1024 * 1024 * 1024L;
+  static final long MAX_REPLICA_CAPACITY_IN_BYTES = 10995116277760L; // 10 TB
+
+  /**
+   * Construct and return the instance name given the host and port.
+   * @param host the hostname of the instance.
+   * @param port the port of the instance.
+   * @return the constructed instance name.
+   */
+  public static String getInstanceName(String host, int port) {
+    return port == -1 ? host : host + "_" + port;
+  }
+
+  /**
+   * Parses zk layout JSON string and populates and returns a map of datacenter name to Zk connect strings.
+   * @param zkLayoutJsonString the string containing the Zookeeper layout.
+   * @throws JSONException if there is an error parsing the JSON.
+   */
+  static Map<String, String> parseZkJsonAndPopulateZkInfo(String zkLayoutJsonString) throws JSONException {
+    Map<String, String> dataCenterToZkAddress = new HashMap<>();
+    JSONObject root = new JSONObject(zkLayoutJsonString);
+    JSONArray all = (root).getJSONArray(ZKINFO_STR);
+    for (int i = 0; i < all.length(); i++) {
+      JSONObject entry = all.getJSONObject(i);
+      dataCenterToZkAddress.put(entry.getString(DATACENTER_STR), entry.getString(ZKCONNECTSTR_STR));
+    }
+    return dataCenterToZkAddress;
+  }
+
+  /**
+   * Get the list of sealed replicas on a given instance.
+   * @param instanceConfig the {@link InstanceConfig} associated with the interested instance.
+   * @return the list of sealed replicas.
+   */
+  static List<String> getSealedReplicas(InstanceConfig instanceConfig) {
```

Comment:
I think they do - this class's contract does not say that it is intended to be generic. The practical use is to keep utility methods that are useful more than once. Only requirement is that the method should be static.